src/components/wallet/AccountSelector.tsx

```diff
@@ -1,53 +1,54 @@
-import React, { useState, useEffect, useCallback } from 'react'
+import React, { useCallback } from 'react'
 
+import { Asset } from '@thorchain/asgardex-util'
 import { Menu, Dropdown } from 'antd'
 
+import { Size as CoinSize } from '../../components/uielements/assets/assetIcon/types'
 import Label from '../../components/uielements/label'
-import { shortSymbol } from '../../helpers/tokenHelpers'
-import { UserAssetType } from '../../types/wallet'
-import DynamicCoin from '../shared/icons/DynamicCoin'
+import AssetIcon from '../uielements/assets/assetIcon'
 import { StyledCard, AssetWrapper, AssetInfoWrapper, AssetTitle } from './AccountSelector.style'
 
 // Multi-use 'account selector' component will have a data context
 // Dummy data
-const UserAssets: UserAssetType[] = [
-  { _id: '1', free: 99, frozen: 11, locked: 21, symbol: 'BNB-JST', name: 'Binance', value: 0.99 },
-  { _id: '2', free: 1034, frozen: 38, locked: 101, symbol: 'RUNE-1E0', name: 'Rune', value: 0.25 }
-]
-
-type Props = { symbol?: string; onChange?: Function; size?: string }
-const AccountSelector: React.FC<Props> = ({ symbol, onChange, size }): JSX.Element => {
-  const [symb, setSymb] = useState<string>(symbol || UserAssets[0].symbol)
-  useEffect(() => {
-    if (typeof onChange === 'function') {
-      onChange(symb)
-    }
-  }, [symb, onChange])
+// const UserAssets: UserAssetType[] = [
+//   { _id: '1', free: 99, frozen: 11, locked: 21, asset: 'BNB-JST', name: 'Binance', value: 0.99 },
+//   { _id: '2', free: 1034, frozen: 38, locked: 101, asset: 'RUNE-1E0', name: 'Rune', value: 0.25 }
+// ]
+
+type Props = {
+  asset: Asset
+  assets: Asset[]
+  onChange?: (asset: Asset) => {}
+  size?: CoinSize
+}
+
+const AccountSelector: React.FC<Props> = (props: Props): JSX.Element => {
+  const { asset, assets, onChange = (_) => {}, size = 'normal' } = props
 
   const menu = useCallback(
     () => (
       <Menu>
-        {UserAssets.map((asset: UserAssetType, i: number) => (
-          <Menu.Item key={i} onClick={() => setSymb(asset.symbol)}>
+        {assets.map((asset, i: number) => (
+          <Menu.Item key={i} onClick={() => onChange(asset)}>
```

Comment:
can we use any data from `asset` for keying MenuItem ?

Reply:
Currently all props of `Asset` are optional. However, the best canditate would be `symbol`, which is explicit. The more I think about the `Asset` type, the more it needs to be refactored to have an optional property for `ticker` only. An `Asset` without a `symbol` property is useless. I will check its implementation in `@throchain/asgardex-util` again, it needs to be refactored...

Reply:
^ Changes for `@throchain/asgardex-util` are addressed in https://gitlab.com/thorchain/asgardex-common/asgardex-util/-/issues/9